SimplenoteTests/NSMutableAttributedStringStylingTests.swift

```diff
@@ -24,6 +29,7 @@ class NSMutableAttributedStringStylingTests: XCTestCase {
         let matches = regex.matches(in: string, options: [], range: string.fullRange)
 
         XCTAssertEqual(matches.count, 2)
+        XCTAssertTrue(verifyEveryMatch(in: matches, contains: numberOfExpectedRangesPerMatch))
```

Comment:
would it make sense to choose more behavior-oriented tests vs. testing here the specifics of the implementation?

what's the goal? maybe the tests can demonstrate the intention behind the code instead of being tightly coupled to the implementation. to echo what @aerych wrote, what if we intentionally want to grab the `x` or `X` inside the check and that adds a new range? what if we replace the RegExp matcher for a more robust parsing function? should these tests fail? has the app been broken in any way?

anyway, that's kind of a rant to motivate my principle that tests should only fail when the app and its behaviors are wrong, not when we legitimately make intentional non-breaking changes. we're already strongly coupled here in the test name so that limits our ability to de-couple the tests to some extent, but we don't have to follow suit if we prefer not to.

```swift
func testAttributesChecklistsThatAppearInMiddleOfString() {
  let input = new NSMutableAttachmentString("This is a valid checklist - [ ] Buy avocados - []);
  input.processChecklistsWithColor(new NSColor(…), …, …);

  XCTAssertEqual(input.attributes(at:26), [SPTextAttachment.Key]);
}
```

obviously I don't know `NSMutableAttributedString` very well or how to detect if the attribute/attachment at a given offset is the checklist. my point is wondering how hard it would be to write tests that test for these behaviors which presumably shouldn't change regardless of how we detect the checklists.

Reply:
You know, I wasn't sure about the entire idea of adding a new check to every unit test. 

I've ended up adding a constant in the `NSRegularExpression` extension itself, documented (why it's there, and what for), and added, instead, two specific tests to ensure the RegEx(es) produce always exactly two ranges.

Thanks for stepping by sir!!